sycl/test/extensions/properties/properties_kernel_device_has_macro.cpp

```diff
@@ -0,0 +1,90 @@
+// RUN: %clangxx -fsycl-device-only -S -Xclang -emit-llvm %s -o - | FileCheck %s --check-prefix CHECK-IR
+// RUN: %clangxx -fsycl -Xclang -verify %s
+// expected-no-diagnostics
+
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+using namespace ext::oneapi::experimental;
+
+static constexpr auto device_has_all = device_has<
+    aspect::ext_oneapi_cuda_async_barrier, aspect::ext_oneapi_bfloat16,
+    aspect::custom, aspect::fp16, aspect::fp64, aspect::image,
+    aspect::online_compiler, aspect::online_linker, aspect::queue_profiling,
+    aspect::usm_device_allocations, aspect::usm_restricted_shared_allocations,
+    aspect::usm_system_allocations, aspect::ext_intel_pci_address, aspect::host,
+    aspect::cpu, aspect::gpu, aspect::accelerator,
+    aspect::ext_intel_gpu_eu_count, aspect::ext_intel_gpu_subslices_per_slice,
+    aspect::ext_intel_gpu_eu_count_per_subslice,
+    aspect::ext_intel_max_mem_bandwidth, aspect::ext_intel_mem_channel,
+    aspect::usm_atomic_host_allocations, aspect::usm_atomic_shared_allocations,
+    aspect::atomic64, aspect::ext_intel_device_info_uuid,
+    aspect::ext_oneapi_srgb, aspect::ext_intel_gpu_eu_simd_width,
+    aspect::ext_intel_gpu_slices, aspect::ext_oneapi_native_assert,
+    aspect::host_debuggable, aspect::ext_intel_gpu_hw_threads_per_eu,
+    aspect::usm_host_allocations, aspect::usm_shared_allocations,
+    aspect::ext_intel_free_memory, aspect::ext_intel_device_id>;
+
+// CHECK-IR: spir_func void @{{.*}}Func0{{.*}}(){{.*}} #[[DHAttr1:[0-9]+]]
```

Comment:
I know you mentioned you would add it to the extension doc, but it's not a bad idea to mention the limitation with `__SYCL_EXTERNAL` here. Our dearth of samples often means tests are looked to as canonical examples, so not a bad practice to keep their comments in line with any limitations.

Reply:
I have added another TODO a little further down, together with the one mentioning to add tests once the attributes are correctly applied to SYCL_EXTERNAL functions.